src/smc-webapp/frame-editors/latex-editor/pythontex.ts

```diff
@@ -0,0 +1,102 @@
+/*
+Run PythonTeX
+*/
+
+import { exec, ExecOutput } from "../generic/client";
+import { parse_path } from "../frame-tree/util";
+import { ProcessedLatexLog, Error } from "./latex-log-parser";
+import { BuildLog } from "./actions";
+
+// command documentation
+//
+// we limit the number of jobs, could be bad for memory usage causing OOM or whatnot
+// -j N, --jobs N        Allow N jobs at once; defaults to cpu_count().
+//
+// --rerun={never,modified,errors,warnings,always}
+// This sets the threshold for re-executing code.
+// By default, PythonTEX will rerun code that has been modified or that produced errors on the last run.
+// "always" executes all code always
+
+export async function pythontex(
+  project_id: string,
+  path: string,
+  time: number,
+  force: boolean,
+  status: Function
+): Promise<ExecOutput> {
+  const { base, directory } = parse_path(path);
+  const args = ["--jobs", "2"];
+  if (force) {
+    // forced build implies to run all snippets
+    args.push("--rerun=always");
+  }
+  status(`pythontex ${args.join(" ")}`);
+  const aggregate = time && !force ? { value: time } : undefined;
+  return exec({
+    allow_post: false, // definitely could take a long time to fully run this
+    timeout: 360,
+    bash: true, // timeout is enforced by ulimit
+    command: "pythontex3",
+    args: args.concat(base),
+    project_id: project_id,
+    path: directory,
+    err_on_exit: false,
+    aggregate
+  });
+}
+
+/*
+example of what we're after:
+the line number on the first line is correct (in the tex file)
+
+This is PythonTeX 0.16
+
+----  Messages for py:default:default  ----
+* PythonTeX stderr - error on line 19:
+    File "<outputdir>/py_default_default.py", line 65
+      print(pytex.formatter(34*131*))
+                                   ^
+  SyntaxError: invalid syntax
+
+--------------------------------------------------
+PythonTeX:  pytex-test - 1 error(s), 0 warning(s)
+*/
```

Comment:
above shows how one error looks like. per "session" there is only one error. the parser is quite simple, but at last it correctly extracts the line number